test-data/unit/check-literal.test

```diff
@@ -914,28 +916,31 @@ c: {"a": 1, "b": 2}             # E: Invalid type comment or annotation
 d: {1, 2, 3}                    # E: Invalid type comment or annotation
 
 [case testLiteralDisallowCollections2]
+# flags: --new-semantic-analyzer
 from typing_extensions import Literal
 a: (1, 2, 3)                    # E: Syntax error in type annotation \
                                 # N: Suggestion: Use Tuple[T1, ..., Tn] instead of (T1, ..., Tn)
 b: Literal[[1, 2, 3]]           # E: Parameter 1 of Literal[...] is invalid
-c: [1, 2, 3]                    # E: Invalid type
+c: [1, 2, 3]                    # E: Invalid type: Did you want to use "List[...]"?
```

Comment:
When we give a suggestion, we typically give it as a separate line. For example:

```
x.py:2: error: Expression "[...]" is not a valid type
x.py:2 note: Did you mean "List[t]"? 
```

We could do the same thing for `x: (int, str)` (`Tuple[t1, ..., tN]`), `x: {int}` (`Set[t]`) and `x: {int: str}` (`Dict[kt, vt]`).

Reply:
We already do this for tuples, dict and set are more tricky (and I think more rare) so I would leave this for a separate PR.